bucketeer/src/androidTest/kotlin/io/bucketeer/sdk/android/e2e/BKTClientVariationTest.kt

```diff
@@ -82,6 +85,21 @@ class BKTClientVariationTest {
         reason = BKTEvaluation.Reason.DEFAULT,
       ),
     )
+
+    assertThat(
+      BKTClient.getInstance().stringEvaluationDetails(FEATURE_ID_STRING, defaultValue = "1"),
+    ).isEqualTo(
+      BKTEvaluationDetail(
+        id = "feature-android-e2e-string:4:bucketeer-android-user-id-1",
+        featureId = FEATURE_ID_STRING,
+        featureVersion = 4,
+        userId = USER_ID,
+        variationId = "36a53a17-60b4-4a99-a54a-7fcbf21f7c8c",
+        variationName = "variation 1",
+        variationValue = "value-1",
+        reason = BKTEvaluationDetail.Reason.DEFAULT,
```

Comment:
For unit tests, it would be safer if we had tests to check the default value and the reason, when the feature flag doesn't exist in the cache.
No need to add the same test for e2e tests.

Reply:
I added tests for it 
https://github.com/bucketeer-io/android-client-sdk/pull/180/files#diff-e3063ec01decd033a3d58ff6938582446f6ab91798ce7f7418349e16e98095d0R10

and

https://github.com/bucketeer-io/android-client-sdk/pull/180/files#diff-e3063ec01decd033a3d58ff6938582446f6ab91798ce7f7418349e16e98095d0R10